From 49e98384c00802888a89a7ab2baca0c6491d7f74 Mon Sep 17 00:00:00 2001
From: Josh Soref <jsoref@users.noreply.github.com>
Date: Sun, 27 Jun 2021 03:15:53 -0400
Subject: [PATCH 1/3] Restrict where configuration files can live

* Files must live in the workspace
* Files can't live in git configuration directories
---
 unknown-words.sh | 23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/unknown-words.sh b/unknown-words.sh
index 1d28a91a..1b9cc29d 100755
--- a/unknown-words.sh
+++ b/unknown-words.sh
@@ -373,7 +373,28 @@ check_dictionary() {
 }
 
 cleanup_file() {
-  maybe_bad="$1"
+  export maybe_bad="$1"
+
+  result=0
+  perl -e '
+    use Cwd qw(abs_path);
+    my $maybe_bad=abs_path($ENV{maybe_bad});
+    my $workspace_path=abs_path($ENV{GITHUB_WORKSPACE});
+    if ($maybe_bad !~ /^\Q$workspace_path\E/) {
+      print "::error ::Configuration files must live within $workspace_path...\n";
+      print "::error ::Unfortunately, file $maybe_bad appears to reside elsewhere.\n";
+      exit 3;
+    }
+    if ($maybe_bad =~ m{/\.git/}i) {
+      print "::error ::Configuration files must not live within `.git/`...\n";
+      print "::error ::Unfortunately, file $maybe_bad appears to.\n";
+      exit 4;
+    }
+  ' || result=$?
+  if [ $result -gt 0 ]; then
+    quit $result
+  fi
+
   type="$2"
   case "$type" in
     patterns|excludes|only)

From fe5fb3336de76e36bb215c7d8a243e61b9064936 Mon Sep 17 00:00:00 2001
From: Josh Soref <jsoref@users.noreply.github.com>
Date: Sun, 27 Jun 2021 00:25:50 -0400
Subject: [PATCH 2/3] Only check regular files

We aren't interested in special files.

If a file should be checked, just ensure the file itself is not excluded.
Otherwise, a symlink is just an invitation to complain about the same
file more than once.
---
 spelling-unknown-word-splitter.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/spelling-unknown-word-splitter.pl b/spelling-unknown-word-splitter.pl
index d325e948..67016c14 100755
--- a/spelling-unknown-word-splitter.pl
+++ b/spelling-unknown-word-splitter.pl
@@ -16,9 +16,9 @@
 
 # skip files that don't exist (including dangling symlinks)
 if (scalar @ARGV) {
-  @ARGV = grep {-r || $_ eq '-'} @ARGV;
+  @ARGV = grep {! -l && -f && -r} @ARGV;
   unless (scalar @ARGV) {
-    print STDERR "None of the provided files are readable\n";
+    print STDERR "::warning ::Was not provided any regular readable files\n";
     exit 0;
   }
 }

From 2494e7c6a63001157807f562725a5c80b4281aab Mon Sep 17 00:00:00 2001
From: Josh Soref <jsoref@users.noreply.github.com>
Date: Sun, 27 Jun 2021 22:40:54 -0400
Subject: [PATCH 3/3] Windows is not supported

---
 common.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/common.sh b/common.sh
index c4b0c58a..d06ba743 100755
--- a/common.sh
+++ b/common.sh
@@ -1,5 +1,10 @@
 #!/bin/bash
 if [ "$INITIALIZED" != defined ]; then
+  if [ "$RUNNER_OS" = "Windows" ]; then
+    echo "::error ::Windows isn't currently supported"
+    exit 5
+  fi
+
   now() {
     date +'%s%N'
   }
